managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/EntryImpl.java

```diff
@@ -149,7 +148,15 @@ public long getEntryId() {
 
     @Override
     public int compareTo(EntryImpl other) {
-        return ComparisonChain.start().compare(ledgerId, other.ledgerId).compare(entryId, other.entryId).result();
+        if (this.ledgerId != other.ledgerId) {
+            return this.ledgerId < other.ledgerId ? -1 : 1;
```

Comment:
`java.lang.Long#compare(this.ledgerId, other.ledgerId)` is more readable.

Reply:
It wouldn't be optimal to this case. 

Long.compare has one additional condition:
```
    public static int compare(long x, long y) {
        return (x < y) ? -1 : ((x == y) ? 0 : 1);
    }
```

That's not optimal for the style used in this change.

Reply:
Yep, one more equals check. But i believe it's most likely will be optimized by JIT :)